### PR TITLE
Bug 2034484: feat: bumped library-go

### PR DESCRIPTION
**Pull request by eggfoobar**
Bumping the library-go dependency to latest, this should take advantage of leader election changes for SNO clusters proposed in this [library-go PR](https://github.com/openshift/library-go/pull/1242) and performance improvements in this [PR](https://github.com/openshift/library-go/pull/1235).

Changes:
   - updated library-go
   - update to use default leader election configs based off of cluster topology
   - updated to k8s v0.23.0
   - updated controller-runtime to v0.11.0 to match k8s 23

Signed-off-by: ehila <ehila@redhat.com>

### CODE REVIEW

**Comment by gnufied** [comment]
I think https://github.com/openshift/local-storage-operator/pull/299/files PR also moves all logging in LSO to klog. cc @dobsonj

File: diskmaker/api_updater.go [hunk 1]
```diff
@@ -7,7 +7,6 @@ import (
 
 	localv1 "github.com/openshift/local-storage-operator/api/v1"
 	"github.com/openshift/local-storage-operator/api/v1alpha1"
-	"github.com/prometheus/common/log"
```
[the remark this answers is not in the document]
Comment (dobsonj) [reply diskmaker/api_updater.go:10]: This change looks fine, I think it was a mistake that LSO ever used this prometheus log package.

**Comment by dobsonj** [comment]
@eggfoobar can you create a bugzilla for this?

**Comment by eggfoobar** [comment]
/retitle Bug 2034484: feat: bumped library-go
this PR pulls in the latest library-go and subsequent k8s 0.23.0 bump

**Pull request merged**